colcon_core/task/python/build.py

```diff
@@ -289,6 +299,13 @@ def _symlinks_in_build(self, args, setup_py_data):
             if not os.path.exists(dst):
                 os.symlink(src, dst)
 
+        def unlink_symlinks(symlinks):
+            # Remove symlinks that were only needed during build time
+            for symlink in symlinks:
+                os.unlink(symlink)
+
+        return functools.partial(unlink_symlinks, temp_symlinks)
```

Comment:
I would prefer to keep this simple and just return `temp_symlinks` here and above in the `finally` block loop over the returned symlinks and `unlink()` them.

Reply:
Return iterable and unlink in `finally` instead of returning callable in f5b8bae

Reply:
Overkill in this instance, but another way to defer cleanup like this can be to `ensure_future` on a coroutine that just sits there waiting to be canceled and then performs the cleanup work on cancellation. I've used this approach to do things like clean up a shared database connection when an async webserver exits.